src/app/static/src/app/store/root/getters.ts

```diff
@@ -1,12 +1,63 @@
 import {RootState} from "../../root";
 import {Getter, GetterTree} from "vuex";
+import {Error} from "../../generated"
 
 interface RootGetters {
     isGuest: Getter<RootState, RootState>
 }
 
 export const getters: RootGetters & GetterTree<RootState, RootState> = {
-    isGuest: (state: RootState, getters: any) => {
+    isGuest: (state: RootState) => {
         return state.currentUser == "guest";
+    },
+    errors: (state: RootState) => {
+        const {
+            adr,
+            adrUpload,
+            baseline,
+            downloadResults,
+            load,
+            metadata,
+            modelCalibrate,
+            modelOptions,
+            modelOutput,
+            plottingSelections,
+            projects,
+            surveyAndProgram
+        } = state;
+
+        return ([] as Error[]).concat.apply([] as Error[], [extractErrors(adr),
+            extractErrors(adrUpload),
+            extractErrors(baseline),
+            extractErrors(downloadResults),
+            extractErrors(load),
+            extractErrors(metadata),
+            extractErrors(modelCalibrate),
+            extractErrors(modelOptions),
+            extractErrors(modelOutput),
+            extractErrors(plottingSelections),
+            extractErrors(projects),
+            extractErrors(surveyAndProgram),
+            state.modelRun.errors,
+            state.errors.errors]);
+    }
+}
+
+const extractErrors = (state: any) => {
+    const errors = [] as Error[];
+    extractErrorsRecursively(state, errors);
+    return errors;
+}
+
+const isComplexObject = (state: any) => {
+    return typeof state === 'object' && !Array.isArray(state) && state !== null
+}
+
+const extractErrorsRecursively = (state: any, errors: Error[]) => {
```

Comment:
It's cool that this is recursive, but does it actually need to be at the moment or is it future-proofing? I can't think of any module that doesn't have its errors at the top level. Or is this handing errors properties which are arrays? I can't quite see how it would do that this since `isComplexObject` is ruling out arrays (and also you're handing the array types, `modelRun.errors` and `errors.errors` separately).

Reply:
It's recursive to handle the download results objects, which contain nested errors, see this test: https://github.com/mrc-ide/hint/pull/562/files#diff-c80530016ccbe2b220a5f11a243bed39425fe75f6626fb4ee6b357281710b7e0R81